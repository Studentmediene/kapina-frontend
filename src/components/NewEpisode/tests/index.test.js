import React from 'react';
import { shallow, mount } from 'enzyme';

import { NewEpisode } from '../';

describe('<NewEpisode />', () => {
  const defaultProps = {
    digasBroadcastId: 123,
    id: 123,
    title: 'Episode title',
    lead: 'Episode lead',
    playOnDemand: jest.fn(),
    togglePlayPause: jest.fn(),
    publishAt: '2014-09-08T08:02:17-05:00',
    paused: true,
  };

  it('renders correctly', () => {
    const mockProps = {
      ...defaultProps,
    };
    const tree = shallow(<NewEpisode {...mockProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('should return null on invalid digasBroadcastId', () => {
    const mockProps = {
      ...defaultProps,
      digasBroadcastId: 0,
    };
    const tree = shallow(<NewEpisode {...mockProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('calls playOnDemand when clicked', () => {
    const mockProps = {
      ...defaultProps,
    };
    const tree = mount(<NewEpisode {...mockProps} />);
    tree.find('.playbuttonbox').simulate('click');
    expect(mockProps.playOnDemand).toBeCalled();
  });

  it('calls playOnDemand on keyPress', () => {
    const mockProps = {
      ...defaultProps,
    };
    const tree = mount(<NewEpisode {...mockProps} />);
    tree.find('.playbuttonbox').simulate('keypress');
    expect(mockProps.playOnDemand).toBeCalled();
  });
});
