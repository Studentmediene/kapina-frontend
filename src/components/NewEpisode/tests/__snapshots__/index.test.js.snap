// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewEpisode /> renders correctly 1`] = `
<div
  className="episode"
  onClick={[Function]}
  onKeyPress={[Function]}
>
  <div
    className="playbuttonbox"
  >
    <PlayPauseButton
      black={true}
      paused={true}
    />
  </div>
  <div
    className="meta"
  >
    <div
      className="title"
    >
      Episode title
    </div>
    <div
      className="expandable"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Episode lead",
        }
      }
      style={
        Object {
          "display": "none",
        }
      }
    />
    <div
      className="publishedAt"
    >
      Publisert: 
      08.09.2014
    </div>
  </div>
  <div
    className="expanderbox"
  >
    <Expander
      expandFunction={[Function]}
      expanded={false}
    />
  </div>
</div>
`;

exports[`<NewEpisode /> should return null on invalid digasBroadcastId 1`] = `""`;
